parlametria/crawler/score_ruralistas/analyzer_score_ambientalista.R

```diff
@@ -107,6 +107,107 @@ calcula_score_doacoes_empresas_rurais <- function(
+#' @param doadores_socios_empresas_rurais_2018_datapath Caminho para o dataframe de doações de campanha 
+#' de sócios de empresas rurais em 2018
+#' @param parlamentares_datapath Caminho para o dataframe de parlamentares
+#' @return Dataframe contendo informações dos parlamentares (cpf e id) e proporção de doações de 
+#' empresas agrícolas agroexportadoras em relação ao total doado
+calcula_score_doacoes_empresas_agroexportadoras <- function(
+  doadores_gerais_2018_datapath = here::here("parlametria/raw_data/receitas/deputados_doadores.csv"),
+  doadores_socios_empresas_rurais_2018_datapath = here::here("parlametria/raw_data/empresas/empresas_doadores_agricolas.csv"),
+  parlamentares_datapath = here::here("crawler/raw_data/parlamentares.csv")
+) {
+  library(tidyverse)
+  library(here)
+
+  doadores <- read_csv(doadores_gerais_2018_datapath, col_types = cols(id = "c")) %>% 
+    filter(casa == "camara")
+  
+  lista_doado_por_parlamentar <- doadores %>%
+    group_by(id) %>% 
+    summarise(total_doacao = sum(valor_receita)) %>% 
+    select(id, total_doacao)
+  
+  lista_doadores_agroexportadoras <- read_csv(doadores_socios_empresas_rurais_2018_datapath, col_types = cols(id_deputado = "c")) %>% 
+    filter(exportadora == "sim") %>% 
+    group_by(cpf_cnpj_socio) %>% 
+    summarise(n = n())
+  
+  doadores_agroexportadoras <- doadores %>% 
+    left_join(lista_doadores_agroexportadoras, by = c("cpf_cnpj_doador" = "cpf_cnpj_socio")) %>% 
+    filter(!is.na(n)) %>% 
+    group_by(id) %>% 
+    summarise(total_doacao_agro_exportadora = sum(valor_receita))
+  
+  indice_doadores <- lista_doado_por_parlamentar %>% 
+    inner_join(doadores_agroexportadoras, by = c("id")) %>% 
+    mutate(proporcao_doacoes_agroexportadoras = total_doacao_agro_exportadora/total_doacao)
+  
+  parlamentares <- read_csv(parlamentares_datapath, col_types = cols(id = "c")) %>% 
+    filter(casa == "camara", em_exercicio == 1) %>% 
+    select(id, cpf)
+  
+  parlamentares_doacoes <- parlamentares %>% 
+    left_join(indice_doadores, by = c("id")) %>% 
+    mutate(proporcao_doacoes_agroexportadoras = if_else(is.na(proporcao_doacoes_agroexportadoras), 
+                                                        0, 
+                                                        proporcao_doacoes_agroexportadoras))
+  
+  return(parlamentares_doacoes)
+}
+
+#' @title Recupera lista de empresas agrícolas exportadoras com parlamentares como sócios
+#' @description A partir da lista de empresas agrícolas dos parlamentares filtra para obter apenas as empresas exportadoras.
+#' Dados de exportação segundo o Ministério da Economia.
+#' @param empresas_parlamentares Caminho para o csv com dados de empresas agrícolas com parlamentares como sócios
+#' @return Dataframe com empresas agroexportadoras com parlamentares como sócio
+#' @example parlamentares_socios_agro_exportadoras <- get_empresas_agroexportadoras_parlamentares()
+get_empresas_agroexportadoras_parlamentares <- function(
```

Comment:
Ao executar `parlamentares_socios_agro_exportadoras <- get_empresas_agroexportadoras_parlamentares()` recebo os seguinte erro: 
` Error in .f(.x[[i]], ...) : object 'ano' not found `